test/libsolidity/syntaxTests/pragma/broken_version_1.sol

```diff
@@ -1,3 +1,3 @@
 pragma solidity ^0^1;
```

Comment:
Could you also add a case with a character that is before `'0'` in the sort order?